### PR TITLE
Adds support of custom type parameters (#5)

### PR DESCRIPTION
**Pull request by jgarec**
* Adds new parameters : sourcesCustomTypeMapping &
testsCustomTypeMapping
* Adds one new sample project

### CODE REVIEW

**Comment by jgarec** [comment]
Here's an example of the new parameters : 
```
        <sourcesCustomTypeMapping>
                <customTypeMapping>
                        <type>package body</type>
                        <customMapping>package_bodies</customMapping>
                </customTypeMapping>
        </sourcesCustomTypeMapping>
```

where sourcesCustomTypeMapping contains a list of customTypeMapping

File: utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    
+                    <tests>
+                    	<test>
+                    		<directory>src/test/resources/type-mapping-project/scripts/test</directory>
+							<includes>
+								<include>**/*sql</include>
+							</includes>
+                    	</test>
+                    </tests>
+                    <testsRegexExpression>.*(\\|\/)(\w+)(\\|\/)(\w+)(\\|\/)(\w+).(\w{3})</testsRegexExpression>
```
Comment (jgebal) [inline utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml]: Wirh utPLSQL v3.1 the regex can be of a simplified form. See doc.

File: utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                <version>${pom.version}</version>
+                <goals>
+                    <goal>test</goal>
+                </goals>
+                <configuration>
+                	<!--  Mandatory Attributes -->
+                	
+					<ignoreFailure>false</ignoreFailure>
+					
+					<paths>
```
Comment (jgebal) [inline utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml]: Could we change from `paths` to `suitepaths`? `paths` seem too ambiguous and not self-descriptive enough.

Comment (jgarec) [reply utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml]: Followed in #17

File: utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml (hunk 1 above)
Comment (jgarec) [reply utplsql-maven-plugin/src/test/resources/type-mapping-project/pom.xml]: Could you point the documentation about this ? :)
I need to use the "owner" parameter ?

**Comment by jgebal** [comment]
@alberto-hernandez , @pesse 
Are you OK with merging this PR?

**Comment by pesse** [comment]
Looks like great work. One thing about test showcase: wouldn't it be better/more consistent to integrate that into demo-project and link to that?
I like the package pretty much and it would be a great benefit to have it available for several modules

**Comment by alberto-hernandez** [comment]
Great Work!

**Pull request merged**